### PR TITLE
ENABLE ASAN PACKAGING with component split dev, runtime and asan

### PR DESCRIPTION
**Pull request by arvindcheru**
Proposed Changes:

- Enable ASAN Packaging (component based split for dev, runtime and asan)
- Add LICENSE file to RPP package

### CODE REVIEW

**Comment by kiritigowda** [comment]
@arvindcheru failing CI - http://math-ci.amd.com/blue/organizations/jenkins/main%2FShort-GPUOpen%2Frpp/detail/PR-235/1/pipeline/698

**Comment by kiritigowda** [comment]
@arvindcheru - could you add updates to changelog - Please sync with develop branch.

File: CMakeLists.txt [hunk 1]
```diff
@@ -24,6 +24,13 @@ set(CMAKE_CXX_STANDARD 17)
...
 
+project(rpp VERSION 1.4.0)
+
+if(${ROCM_PATCH_VERSION})
+   set(PROJECT_VERSION_PATCH ${ROCM_PATCH_VERSION})
+   set(PROJECT_VERSION "${PROJECT_VERSION_MAJOR}.${PROJECT_VERSION_MINOR}.${PROJECT_VERSION_PATCH}")
+endif()
+
 # RPP Version
 set(VERSION "1.4.0")
```
Comment (kiritigowda) [inline CMakeLists.txt:35]: @arvindcheru can we use the version info from one place?

Comment (arvindcheru) [reply CMakeLists.txt:26]: Updated in latest version

**Comment by arvindcheru** [comment]
> @arvindcheru - could you add updates to changelog - Please sync with develop branch.

Updated Changelog Pls review

**Comment by kiritigowda** [comment]
@arvindcheru - merge pending CI - failing - http://math-ci.amd.com/blue/organizations/jenkins/main%2FShort-GPUOpen%2Frpp/detail/PR-235/1/pipeline 
Can you please take a look?

**Comment by kiritigowda** [comment]
@arvindcheru - CI Pass - http://math-ci.amd.com/blue/organizations/jenkins/main%2FShort-GPUOpen%2Frpp/detail/PR-235/4/pipeline/924

**Pull request merged**